File: features/lead/src/main/java/com/nivasafinance/features/leadlender/service/LeadLenderReadService.java
package com.nivasafinance.features.leadlender.service;

import com.nivasafinance.features.leadlender.dto.LeadLenderResponse;

import java.util.List;
import java.util.UUID;

public interface LeadLenderReadService {
    
    List<LeadLenderResponse> getLeadLenders(UUID leadIdentifier, List<String> status);
    
    LeadLenderResponse getLeadLenderByIdentifier(UUID leadIdentifier, UUID lenderIdentifier);
}

File: features/lead/src/main/java/com/nivasafinance/features/leadlender/service/LeadLenderReadService.kt
package com.nivasafinance.features.leadlender.service

import com.nivasafinance.features.leadlender.dto.LeadLenderResponse
import java.util.UUID

interface LeadLenderReadService {
    fun getLeadLenders(leadId: UUID): List<LeadLenderResponse>
    fun getLeadLenderById(leadLenderId: UUID): LeadLenderResponse
}
